Rename CSSParserSelector to MutableCSSSelector

nt1m opened the pull request:
#### ac784c090ce396d3b17092ca7ad4ac8968c48f8f
<pre>
Rename CSSParserSelector to MutableCSSSelector
<a href="https://bugs.webkit.org/show_bug.cgi?id=267037">https://bugs.webkit.org/show_bug.cgi?id=267037</a>
<a href="https://rdar.apple.com/120412070">rdar://120412070</a>

Reviewed by Antti Koivisto.

The current name is confusing, especially with the presence of CSSSelectorParser. Rename to MutableCSSSelector to actually represent its purpose.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSSelector.cpp:
* Source/WebCore/css/CSSSelectorList.cpp:
(WebCore::CSSSelectorList::CSSSelectorList):
* Source/WebCore/css/CSSSelectorList.h:
* Source/WebCore/css/StyleRule.cpp:
* Source/WebCore/css/parser/MutableCSSSelector.cpp: Renamed from Source/WebCore/css/parser/CSSParserSelector.cpp.
(WebCore::MutableCSSSelector::parsePagePseudoSelector):
(WebCore::MutableCSSSelector::parsePseudoElementSelector):
(WebCore::MutableCSSSelector::parsePseudoClassSelector):
(WebCore::MutableCSSSelector::MutableCSSSelector):
(WebCore::MutableCSSSelector::~MutableCSSSelector):
(WebCore::MutableCSSSelector::adoptSelectorVector):
(WebCore::MutableCSSSelector::setArgumentList):
(WebCore::MutableCSSSelector::setSelectorList):
(WebCore::MutableCSSSelector::leftmostSimpleSelector const):
(WebCore::MutableCSSSelector::leftmostSimpleSelector):
(WebCore::MutableCSSSelector::hasExplicitNestingParent const):
(WebCore::MutableCSSSelector::hasExplicitPseudoClassScope const):
(WebCore::selectorListMatchesPseudoElement):
(WebCore::MutableCSSSelector::matchesPseudoElement const):
(WebCore::MutableCSSSelector::insertTagHistory):
(WebCore::MutableCSSSelector::appendTagHistory):
(WebCore::MutableCSSSelector::appendTagHistoryAsRelative):
(WebCore::MutableCSSSelector::prependTagSelector):
(WebCore::MutableCSSSelector::releaseTagHistory):
(WebCore::MutableCSSSelector::isHostPseudoSelector const):
(WebCore::MutableCSSSelector::startsWithExplicitCombinator const):
* Source/WebCore/css/parser/MutableCSSSelector.h: Renamed from Source/WebCore/css/parser/CSSParserSelector.h.
(WebCore::MutableCSSSelector::releaseSelector):
(WebCore::MutableCSSSelector::selector const):
(WebCore::MutableCSSSelector::selector):
(WebCore::MutableCSSSelector::setValue):
(WebCore::MutableCSSSelector::setAttribute):
(WebCore::MutableCSSSelector::setArgument):
(WebCore::MutableCSSSelector::setNth):
(WebCore::MutableCSSSelector::setMatch):
(WebCore::MutableCSSSelector::setRelation):
(WebCore::MutableCSSSelector::setForPage):
(WebCore::MutableCSSSelector::match const):
(WebCore::MutableCSSSelector::pseudoElement const):
(WebCore::MutableCSSSelector::selectorList const):
(WebCore::MutableCSSSelector::setPseudoElement):
(WebCore::MutableCSSSelector::setPseudoClass):
(WebCore::MutableCSSSelector::pseudoClass const):
(WebCore::MutableCSSSelector::tagHistory const):
(WebCore::MutableCSSSelector::setTagHistory):
(WebCore::MutableCSSSelector::clearTagHistory):
(WebCore::MutableCSSSelector::needsImplicitShadowCombinatorForMatching const):
* Source/WebCore/css/parser/CSSParserImpl.cpp:
(WebCore::appendImplicitSelectorPseudoClassScopeIfNeeded):
(WebCore::appendImplicitSelectorNestingParentIfNeeded):
(WebCore::CSSParserImpl::appendImplicitSelectorIfNeeded):
(WebCore::CSSParserImpl::parsePageSelector):
(WebCore::CSSParserImpl::createNestingParentRule):
(WebCore::CSSParserImpl::consumeScopeRule):
(WebCore::CSSParserImpl::consumeStyleRule):
* Source/WebCore/css/parser/CSSParserImpl.h:
* Source/WebCore/css/parser/CSSSelectorParser.cpp:
(WebCore::parseMutableCSSSelectorList):
(WebCore::parseCSSSelectorList):
(WebCore::CSSSelectorParser::consumeSelectorList):
(WebCore::CSSSelectorParser::consumeComplexSelectorList):
(WebCore::CSSSelectorParser::consumeRelativeSelectorList):
(WebCore::CSSSelectorParser::consumeNestedSelectorList):
(WebCore::CSSSelectorParser::consumeForgivingSelectorList):
(WebCore::CSSSelectorParser::consumeComplexForgivingSelectorList):
(WebCore::CSSSelectorParser::consumeNestedComplexForgivingSelectorList):
(WebCore::CSSSelectorParser::supportsComplexSelector):
(WebCore::CSSSelectorParser::consumeCompoundSelectorList):
(WebCore::extractCompoundFlags):
(WebCore::CSSSelectorParser::consumeNestedComplexSelector):
(WebCore::CSSSelectorParser::consumeComplexSelector):
(WebCore::CSSSelectorParser::consumeRelativeScopeSelector):
(WebCore::CSSSelectorParser::consumeRelativeNestedSelector):
(WebCore::isSimpleSelectorValidAfterPseudoElement):
(WebCore::CSSSelectorParser::consumeCompoundSelector):
(WebCore::CSSSelectorParser::consumeSimpleSelector):
(WebCore::CSSSelectorParser::consumeId):
(WebCore::CSSSelectorParser::consumeClass):
(WebCore::CSSSelectorParser::consumeNesting):
(WebCore::CSSSelectorParser::consumeAttribute):
(WebCore::CSSSelectorParser::consumePseudo):
(WebCore::CSSSelectorParser::prependTypeSelectorIfNeeded):
(WebCore::CSSSelectorParser::splitCompoundAtImplicitShadowCrossingCombinator):
(WebCore::CSSSelectorParser::resolveNestingParent):
(WebCore::parseCSSParserSelectorList): Deleted.
* Source/WebCore/css/parser/CSSSelectorParser.h:
* Source/WebCore/css/process-css-pseudo-selectors.py:

Canonical link: <a href="https://commits.webkit.org/272620@main">https://commits.webkit.org/272620@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ab09edaa1ac414b33362459043d4f62fbec70d9e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32409 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11150 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34241 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/34968 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/29334 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/33264 "Passed tests") | [❌ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13503 "Failed to checkout and rebase branch from PR 22350") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/51/builds/8354 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/28867 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/32833 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/49/builds/13503 "Failed to checkout and rebase branch from PR 22350") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/28971 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/8185 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/49/builds/13503 "Failed to checkout and rebase branch from PR 22350") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/28902 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/36306 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/49/builds/13503 "Failed to checkout and rebase branch from PR 22350") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/29332 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/36306 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/8463 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/51/builds/8354 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/36306 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/10124 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/9084 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4188 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/9033 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/CSSSelectorList.cpp:
```diff
@@ -44,21 +44,21 @@ CSSSelectorList::CSSSelectorList(const CSSSelectorList& other)
     size_t flattenedSize = 0;
     for (size_t i = 0; i < selectorVector.size(); ++i) {
-        for (CSSParserSelector* selector = selectorVector[i].get(); selector; selector = selector->tagHistory())
+        for (MutableCSSSelector* selector = selectorVector[i].get(); selector; selector = selector->tagHistory())
```
Could probably use `auto*`

nt1m commented on Source/WebCore/css/CSSSelectorList.cpp:
```diff
@@ -44,21 +44,21 @@ CSSSelectorList::CSSSelectorList(const CSSSelectorList& other)
-        CSSParserSelector* first = selectorVector[i].get();
-        CSSParserSelector* current = first;
+        MutableCSSSelector* first = selectorVector[i].get();
+        MutableCSSSelector* current = first;
```
auto*

nt1m commented on Source/WebCore/css/MutableCSSSelector.cpp:
```diff
@@ -232,34 +232,34 @@ void CSSParserSelector::appendTagHistoryAsRelative(std::unique_ptr<CSSParserSele
+void MutableCSSSelector::appendTagHistory(Combinator relation, std::unique_ptr<MutableCSSSelector> selector)
 {
-    CSSParserSelector* end = this;
+    MutableCSSSelector* end = this;
```
auto*

nt1m commented on Source/WebCore/css/parser/CSSSelectorParser.cpp:
```diff
@@ -108,39 +108,39 @@ CSSParserSelectorList CSSSelectorParser::consumeSelectorList(CSSParserTokenRange
         return { };
 
-    Vector<std::unique_ptr<CSSParserSelector>> selectorList;
+    Vector<std::unique_ptr<MutableCSSSelector>> selectorList;
```
MutableCSSSelectorList selectorList;

nt1m commented on Source/WebCore/css/parser/CSSSelectorParser.cpp:
```diff
@@ -228,9 +228,9 @@ bool CSSSelectorParser::supportsComplexSelector(CSSParserTokenRange range, const
+MutableCSSSelectorList CSSSelectorParser::consumeCompoundSelectorList(CSSParserTokenRange& range)
 {
-    Vector<std::unique_ptr<CSSParserSelector>> selectorList;
+    Vector<std::unique_ptr<MutableCSSSelector>> selectorList;
```
MutableCSSSelectorList selectorList;

nt1m commented on Source/WebCore/css/parser/CSSSelectorParser.cpp:
```diff
@@ -311,15 +311,15 @@ std::unique_ptr<CSSParserSelector> CSSSelectorParser::consumeNestedComplexSelect
     OptionSet<CompoundSelectorFlag> previousCompoundFlags;
 
-    for (CSSParserSelector* simple = selector.get(); simple && !previousCompoundFlags; simple = simple->tagHistory())
+    for (MutableCSSSelector* simple = selector.get(); simple && !previousCompoundFlags; simple = simple->tagHistory())
```
auto*

nt1m commented on Source/WebCore/css/parser/CSSSelectorParser.cpp:
```diff
@@ -332,7 +332,7 @@ std::unique_ptr<CSSParserSelector> CSSSelectorParser::consumeComplexSelector(CSS
         if (previousCompoundFlags.contains(CompoundSelectorFlag::HasPseudoElementForRightmostCompound))
             return nullptr;
-        CSSParserSelector* end = nextSelector.get();
+        MutableCSSSelector* end = nextSelector.get();
```
auto*

nt1m commented on Source/WebCore/css/parser/CSSSelectorParser.cpp:
```diff
@@ -1148,7 +1148,7 @@ std::unique_ptr<CSSParserSelector> CSSSelectorParser::splitCompoundAtImplicitSha
     // Example: input#x::-webkit-inner-spin-button -> [ ::-webkit-inner-spin-button, input, #x ]
     //
-    CSSParserSelector* splitAfter = compoundSelector.get();
+    MutableCSSSelector* splitAfter = compoundSelector.get();
```
auto*

nt1m commented on Source/WebCore/css/parser/CSSSelectorParser.cpp:
```diff
@@ -1200,18 +1200,18 @@ bool CSSSelectorParser::containsUnknownWebKitPseudoElements(const CSSSelector& c
 CSSSelectorList CSSSelectorParser::resolveNestingParent(const CSSSelectorList& nestedSelectorList, const CSSSelectorList* parentResolvedSelectorList)
 {
-    Vector<std::unique_ptr<CSSParserSelector>> result;
+    Vector<std::unique_ptr<MutableCSSSelector>> result;
```
MutableSelectorList result;

nt1m reviewed (commented):
Notes to self

anttijk commented on Source/WebCore/Headers.cmake:
```diff
@@ -878,7 +879,6 @@ set(WebCore_PRIVATE_FRAMEWORK_HEADERS
     css/parser/CSSParserContext.h
     css/parser/CSSParserEnum.h
     css/parser/CSSParserMode.h
-    css/parser/CSSParserSelector.h
```
Not sure this needs to move. It is still primarily a parser thing.

nt1m committed:
Rename CSSParserSelector to MutableCSSSelector
https://bugs.webkit.org/show_bug.cgi?id=267037
rdar://120412070

Reviewed by Antti Koivisto.

The current name is confusing, especially with the presence of CSSSelectorParser. Rename to MutableCSSSelector to actually represent its purpose.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSSelector.cpp:
* Source/WebCore/css/CSSSelectorList.cpp:
(WebCore::CSSSelectorList::CSSSelectorList):
* Source/WebCore/css/CSSSelectorList.h:
* Source/WebCore/css/StyleRule.cpp:
* Source/WebCore/css/parser/MutableCSSSelector.cpp: Renamed from Source/WebCore/css/parser/CSSParserSelector.cpp.
(WebCore::MutableCSSSelector::parsePagePseudoSelector):
(WebCore::MutableCSSSelector::parsePseudoElementSelector):
(WebCore::MutableCSSSelector::parsePseudoClassSelector):
(WebCore::MutableCSSSelector::MutableCSSSelector):
(WebCore::MutableCSSSelector::~MutableCSSSelector):
(WebCore::MutableCSSSelector::adoptSelectorVector):
(WebCore::MutableCSSSelector::setArgumentList):
(WebCore::MutableCSSSelector::setSelectorList):
(WebCore::MutableCSSSelector::leftmostSimpleSelector const):
(WebCore::MutableCSSSelector::leftmostSimpleSelector):
(WebCore::MutableCSSSelector::hasExplicitNestingParent const):
(WebCore::MutableCSSSelector::hasExplicitPseudoClassScope const):
(WebCore::selectorListMatchesPseudoElement):
(WebCore::MutableCSSSelector::matchesPseudoElement const):
(WebCore::MutableCSSSelector::insertTagHistory):
(WebCore::MutableCSSSelector::appendTagHistory):
(WebCore::MutableCSSSelector::appendTagHistoryAsRelative):
(WebCore::MutableCSSSelector::prependTagSelector):
(WebCore::MutableCSSSelector::releaseTagHistory):
(WebCore::MutableCSSSelector::isHostPseudoSelector const):
(WebCore::MutableCSSSelector::startsWithExplicitCombinator const):
* Source/WebCore/css/parser/MutableCSSSelector.h: Renamed from Source/WebCore/css/parser/CSSParserSelector.h.
(WebCore::MutableCSSSelector::releaseSelector):
(WebCore::MutableCSSSelector::selector const):
(WebCore::MutableCSSSelector::selector):
(WebCore::MutableCSSSelector::setValue):
(WebCore::MutableCSSSelector::setAttribute):
(WebCore::MutableCSSSelector::setArgument):
(WebCore::MutableCSSSelector::setNth):
(WebCore::MutableCSSSelector::setMatch):
(WebCore::MutableCSSSelector::setRelation):
(WebCore::MutableCSSSelector::setForPage):
(WebCore::MutableCSSSelector::match const):
(WebCore::MutableCSSSelector::pseudoElement const):
(WebCore::MutableCSSSelector::selectorList const):
(WebCore::MutableCSSSelector::setPseudoElement):
(WebCore::MutableCSSSelector::setPseudoClass):
(WebCore::MutableCSSSelector::pseudoClass const):
(WebCore::MutableCSSSelector::tagHistory const):
(WebCore::MutableCSSSelector::setTagHistory):
(WebCore::MutableCSSSelector::clearTagHistory):
(WebCore::MutableCSSSelector::needsImplicitShadowCombinatorForMatching const):
* Source/WebCore/css/parser/CSSParserImpl.cpp:
(WebCore::appendImplicitSelectorPseudoClassScopeIfNeeded):
(WebCore::appendImplicitSelectorNestingParentIfNeeded):
(WebCore::CSSParserImpl::appendImplicitSelectorIfNeeded):
(WebCore::CSSParserImpl::parsePageSelector):
(WebCore::CSSParserImpl::createNestingParentRule):
(WebCore::CSSParserImpl::consumeScopeRule):
(WebCore::CSSParserImpl::consumeStyleRule):
* Source/WebCore/css/parser/CSSParserImpl.h:
* Source/WebCore/css/parser/CSSSelectorParser.cpp:
(WebCore::parseMutableCSSSelectorList):
(WebCore::parseCSSSelectorList):
(WebCore::CSSSelectorParser::consumeSelectorList):
(WebCore::CSSSelectorParser::consumeComplexSelectorList):
(WebCore::CSSSelectorParser::consumeRelativeSelectorList):
(WebCore::CSSSelectorParser::consumeNestedSelectorList):
(WebCore::CSSSelectorParser::consumeForgivingSelectorList):
(WebCore::CSSSelectorParser::consumeComplexForgivingSelectorList):
(WebCore::CSSSelectorParser::consumeNestedComplexForgivingSelectorList):
(WebCore::CSSSelectorParser::supportsComplexSelector):
(WebCore::CSSSelectorParser::consumeCompoundSelectorList):
(WebCore::extractCompoundFlags):
(WebCore::CSSSelectorParser::consumeNestedComplexSelector):
(WebCore::CSSSelectorParser::consumeComplexSelector):
(WebCore::CSSSelectorParser::consumeRelativeScopeSelector):
(WebCore::CSSSelectorParser::consumeRelativeNestedSelector):
(WebCore::isSimpleSelectorValidAfterPseudoElement):
(WebCore::CSSSelectorParser::consumeCompoundSelector):
(WebCore::CSSSelectorParser::consumeSimpleSelector):
(WebCore::CSSSelectorParser::consumeId):
(WebCore::CSSSelectorParser::consumeClass):
(WebCore::CSSSelectorParser::consumeNesting):
(WebCore::CSSSelectorParser::consumeAttribute):
(WebCore::CSSSelectorParser::consumePseudo):
(WebCore::CSSSelectorParser::prependTypeSelectorIfNeeded):
(WebCore::CSSSelectorParser::splitCompoundAtImplicitShadowCrossingCombinator):
(WebCore::CSSSelectorParser::resolveNestingParent):
(WebCore::parseCSSParserSelectorList): Deleted.
* Source/WebCore/css/parser/CSSSelectorParser.h:
* Source/WebCore/css/process-css-pseudo-selectors.py:

Canonical link: https://commits.webkit.org/272620@main
